src/scheduler/node_info.c

```diff
@@ -1693,17 +1693,17 @@ update_node_on_run(nspec *ns, resource_resv *resresv, char *job_state)
 			}
 		}
 	}
-	
+
 	for(te = resresv->server->calendar->next_event; te != NULL; te = te->next) {
 		if(te->event_type & TIMED_RUN_EVENT) {
 			if(te->event_ptr == resresv)
 				break;
 		}
 	}
-	
+
 	remove_te_list(&ninfo->node_events, te);
 
-	if (ninfo->bucket_ind != -1) {
+	if (ninfo->node_ind != 1 && ninfo->bucket_ind != -1) {
```

Comment:
Shouldn't that be node_ind != -1?

Reply:
I had this weird feeling that something was not quite right there :)